io.openems.edge.ess.byd.container/src/io/openems/edge/ess/byd/container/watchdog/BydContainerWatchdog.java

```diff
@@ -0,0 +1,146 @@
+package io.openems.edge.ess.byd.container.watchdog;
+
+import java.util.Optional;
+
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.openems.common.channel.AccessMode;
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.types.OpenemsType;
+import io.openems.edge.common.channel.Doc;
+import io.openems.edge.common.channel.IntegerWriteChannel;
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.ComponentManager;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.common.modbusslave.ModbusSlave;
+import io.openems.edge.common.modbusslave.ModbusSlaveNatureTable;
+import io.openems.edge.common.modbusslave.ModbusSlaveTable;
+import io.openems.edge.common.modbusslave.ModbusType;
+import io.openems.edge.controller.api.Controller;
+import io.openems.edge.ess.byd.container.EssFeneconBydContainer;
+import io.openems.edge.ess.power.api.Phase;
+import io.openems.edge.ess.power.api.Pwr;
+import io.openems.edge.ess.power.api.Relationship;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Ess.Fenecon.BydContainer.WatchdogController", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE)
+public class BydContainerWatchdog extends AbstractOpenemsComponent
+		implements Controller, OpenemsComponent, ModbusSlave {
+
+	private final Logger log = LoggerFactory.getLogger(BydContainerWatchdog.class);
+
+	@Reference
+	public ComponentManager componentManager;
+
+	@Reference
+	protected ConfigurationAdmin cm;
+
+	public Config config;
```

Comment:
Does this need to be public? If yes, a getter method would be preferrable to avoid someone overwriting the variable.